Lib/_pyrepl/reader.py

```diff
@@ -240,6 +240,8 @@ class Reader:
     lxy: tuple[int, int] = field(init=False)
     calc_screen: CalcScreen = field(init=False)
     scheduled_commands: list[str] = field(default_factory=list)
+    can_colorize: bool = False
```

Comment:
This is an optimisation as this call has a non-trivial cost for showing prompts

Reply:
Can this change while the REPL is alive?

Reply:
Yes but is unlikely and would be an implementation detail in any case. For instance in the trace back module we cache the first result to not call it all the time